jdbc/src/main/java/org/jclouds/jdbc/strategy/JdbcStorageStrategy.java

```diff
@@ -123,63 +121,82 @@ public void setContainerAccess(String container, ContainerAccess containerAccess
    @Override
    public void deleteContainer(String container) {
       jdbcContainerNameValidator.validate(container);
+      jdbcService.deleteBlobsByContainer(container);
       jdbcService.deleteContainerByName(container);
    }
 
    @Override
-   public void clearContainer(String s) {
-
+   public void clearContainer(String container) {
+      jdbcService.deleteBlobsByContainer(container);
    }
 
    @Override
-   public void clearContainer(String s, ListContainerOptions listContainerOptions) {
+   public void clearContainer(String container, ListContainerOptions options) {
+      if (options.getDir() != null) {
+         jdbcService.deleteBlobsByDirectory(container, options.getDir(), options.isRecursive());
+         return;
+      }
 
+      clearContainer(container);
    }
 
    @Override
-   @Transactional
-   public StorageMetadata getContainerMetadata(String container) {
-      MutableStorageMetadata metadata = new MutableStorageMetadataImpl();
-      metadata.setName(container);
-      metadata.setType(StorageType.CONTAINER);
-      metadata.setLocation(null);
-      metadata.setCreationDate(jdbcService.findContainerByName(container).getCreationDate());
+   public StorageMetadata getContainerMetadata(String containerName) {
+      ContainerEntity containerEntity = jdbcService.findContainerByName(containerName);
+      MutableStorageMetadata metadata = null;
+      if (containerEntity != null) {
+         metadata = new MutableStorageMetadataImpl();
+         metadata.setName(containerName);
+         metadata.setType(StorageType.CONTAINER);
+         metadata.setLocation(null);
+         metadata.setCreationDate(containerEntity.getCreationDate());
+      }
       return metadata;
    }
 
    @Override
-   public boolean blobExists(String s, String s1) {
-      return false;
+   public boolean blobExists(String container, String key) {
```

Comment:
At least the parameters in the strategy methods should probably have javadocs, even if the methods are mostly self-documented. Other stuff that might be useful is null-safety or invalid characters, if any.
The strategy itself probably needs some javadocs on how stuff is stored.